### ISSUE TITLE
bug: OpenSearch call fails with https

### ISSUE THREAD

**Issue by eddumelendez** [localstack/localstack#9824]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

Execute `http https://testcontainers-domain.us-east-1.opensearch.localhost.localstack.cloud:4566` after creating a OpenSearch domain returns

```
http: error: SSLError: HTTPSConnectionPool(host='testcontainers-domain.us-east-1.opensearch.localhost.localstack.cloud', port=4566): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, "[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'testcontainers-domain.us-east-1.opensearch.localhost.localstack.cloud'. (_ssl.c:1000)"))) while doing a GET request to URL: https://testcontainers-domain.us-east-1.opensearch.localhost.localstack.cloud:4566/
```

### Expected Behavior

A successful response and similar output as `http http://testcontainers.us-east-1.opensearch.localhost.localstack.cloud:4566`

```
HTTP/1.1 200
content-encoding: gzip
content-length: 338
content-type: application/json; charset=UTF-8
date: Fri, 08 Dec 2023 03:42:05 GMT
server: hypercorn-h11

{
    "cluster_name": "opensearch",
    "cluster_uuid": "RjqsUG8pTTyFEdrrMcG1tw",
    "name": "2aeeda41f987",
    "tagline": "The OpenSearch Project: https://opensearch.org/",
    "version": {
        "build_date": "2023-07-18T21:22:48.164885046Z",
        "build_hash": "1164221ee2b8ba3560f0ff492309867beea28433",
        "build_snapshot": false,
        "build_type": "tar",
        "distribution": "opensearch",
        "lucene_version": "9.7.0",
        "minimum_index_compatibility_version": "7.0.0",
        "minimum_wire_compatibility_version": "7.10.0",
        "number": "2.9.0"
    }
}
```

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

```bash
docker run -p4566:4566 localstack/localstack:3.0.2
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```bash
docker exec -ti <container-id> /bin/bash

awslocal opensearch create-domain --domain-name testcontainers-domain

awslocal opensearch describe-domain --domain-name testcontainers-domain

# Wait until .DomainStatus.Processing is equal to false and exit the container
```

### Environment

```markdown
- OS: macos Sonoma 14.1.2
- LocalStack: 3.0.2
```

**Comment by eddumelendez** [comment]
The reason about this issue is because I'm trying to contribute to LangChain4j project which uses AWS OpenSearch API as one of the clients and it forces to use [https](https://github.com/opensearch-project/opensearch-java/blob/ef04bf87b3557e6f090e41859a8c356a111224e2/java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java#L287). You can see my branch [here](https://github.com/eddumelendez/langchain4j/blob/testcontainers-localstack/langchain4j-opensearch/src/test/java/dev/langchain4j/store/embedding/opensearch/OpenSearchEmbeddingStoreAWSTest.java).

I think this is a good opportunity to show how test those applications using LocalStack and Testcontainers.

**Comment by eddumelendez** [comment]
Fixed in LS 3.1
